Drop type specs for Trigger

NiharSaha opened the pull request:
#### PR description:

Drop type specs in DQMOffline/Trigger python configuration files.

Central DQM migration campaign for drop type specs following
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1
and update the safer syntax for existing parameters like: (1) drop type specifications where the original parameter exists. (2) using "clone" instead of "deepcopy" (3) move all parameters inside the clone

MET hanlde not valid" message now disappears, this was due to a wrong collection set BEFORE the PR, since Event Interpretation has been removed back in 2021.
This PR corrected above by setting to met = "pfMet", right above. Since the SoftMuHardJetMETSUSYmonitoring is cloned several times below, it seems it just complained but no crash. Now it should be reading an existing collection.

Moreover, the binning for diphotonMass55ANDnoPV module was corrected set to 50-150 in https://github.com/cms-sw/cmssw/pull/37133/files#diff-f0c298feb52e2a455f0a07a4ea80c12b3b660e38e6737d177570ccc9a94775d3R165

since it was wrongly set to 90-200 due to a typo in https://github.com/cms-sw/cmssw/pull/37133/files#diff-f0c298feb52e2a455f0a07a4ea80c12b3b660e38e6737d177570ccc9a94775d3L147

#### PR validation:
Tested in CMSSW_12_3_X via runTheMatrix.py -l limited -i all --ibeos

NiharSaha committed:
Drop type specs for Trigger

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37133/28666

- This PR adds an extra 16KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Trigger/python/B2GMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/B2GTnPMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BPHMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BPHMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BTVHLTOfflineSource_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BTaggingMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BTaggingMonitoring_Client_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/BTaggingMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DQMOffline_LumiMontiroring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DQMOffline_Trigger_cosmics_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DiDispStaMuonMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DiJetMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DiJetMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/DisplacedJet_Monitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/EgHLTOfflineSource_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/FSQHLTOfflineSource_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HILowLumiHLTOfflineSource_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HLTMuonOfflineAnalyzer_cosmics_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HLTTauDQMOffline_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HMesonGammaMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HTMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HTMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HiggsMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/HiggsMonitoring_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/JetMETHLTOfflineClient_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/JetMETHLTOfflineSource_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/JetMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/JetMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/LepHTMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/METMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/METMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/METplusTrackMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/METplusTrackMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MjjMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MssmHbbBtagTriggerMonitor_Client_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MssmHbbBtagTriggerMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MssmHbbMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MssmHbbMonitoring_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MuonMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/MuonMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/NoBPTXMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/NoBPTXMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/ObjMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/PhotonMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/PhotonMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/PrimaryVertexMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/RazorMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/RazorMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_HistogramManager_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SiStrip_OfflineMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SoftMuHardJetMETSUSYMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SoftdropMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SusyMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/SusyMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/Tau3MuMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/Tau3MuMonitor_cfi.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TopMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TrackToTrackMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TrackingMonitoringPA_Client_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TrackingMonitoringPA_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TrackingMonitoring_Client_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/TrackingMonitoring_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/VBFMETMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/VBFSUSYMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/VBFTauMonitor_cff.py  modified in PR(s): #36783
  - File  DQMOffline/Trigger/python/susyHLTEleCaloJets_cff.py  modified in PR(s): #36783

cmsbuild commented:
A new Pull Request was created by  @NiharSaha (Nihar Ranjan Saha) for master.

It involves the following packages:

- DQMOffline/Trigger (**dqm**)


@emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @pbo0, @rvenditti can you please review it and eventually sign? Thanks.
@sscruz, @missirol, @mtosi, @Fedespring, @calderona, @HuguesBrun, @jhgoh, @trocino, @cericeci, @rociovilar this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

jfernan2 commented:
This PR replaces https://github.com/cms-sw/cmssw/pull/36783  squashing all commits into a single one

jfernan2 commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f581ae/22811/summary.html
**COMMIT**: 4facfebaeed85d6a77e86366deb4af8cb882e391
**CMSSW**: CMSSW_12_3_X_2022-03-03-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37133/22811/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 24 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3987741
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 4
 - DQMHistoTests: Total successes: 3987707
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 45 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-03-03-1100+f581ae/48700/triggerResults): no differences found

jfernan2 commented:
+1
@NiharSaha do not forget to close https://github.com/cms-sw/cmssw/pull/36783

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

NiharSaha commented:
> +1 @NiharSaha do not forget to close #36783

Yes @jfernan2. I have closed the PR #36783. Thank you.

perrotta commented:
I'm ready to merge this, but out, of curiosity: what was the fix that allowed to remove from output the error message
```
%MSG-w TopMonitor:  TopMonitor:SoftMuHardJetMETSUSYmonitoring_metPt  03-Mar-2022 12:40:15 CET Run: 1 Event: 3
MET handle not valid 

%MSG
%MSG-w TopMonitor:  TopMonitor:SoftMuHardJetMETSUSYmonitoring_jetPt  03-Mar-2022 12:40:15 CET Run: 1 Event: 3
MET handle not valid 

%MSG
```

I cannot see any obvious difference in practice in the configuration of `SoftMuHardJetMETSUSYmonitoring_jetPt` in the baseline and in this PR, but I am certainly missing something. (That was probably already discussed in https://github.com/cms-sw/cmssw/pull/36783, but I did not manage to scroll through the whole long thread in that closed PR).

PS: if a bug fix is applied, discovered thanks to this campaign, it is proably worth to write it in the PR description

jfernan2 commented:
Hi @perrotta 
Yes, this was commented in: https://github.com/cms-sw/cmssw/pull/36783#discussion_r811226931

@NiharSaha please include the following in the PR description:

"MET hanlde not valid" message now disappears, this was due to a wrong collection set BEFORE the PR, since Event Interpretation has been removed back in 2021.
This PR corrected above by setting to met = "pfMet", right above. Since the SoftMuHardJetMETSUSYmonitoring is cloned several times below, it seems it just complained but no crash. Now it should be reading an existing collection.

Moreover, the binning for diphotonMass55ANDnoPV module was corrected set to 50-150 in https://github.com/cms-sw/cmssw/pull/37133/files#diff-f0c298feb52e2a455f0a07a4ea80c12b3b660e38e6737d177570ccc9a94775d3R165

since it was wrongly set to 90-200 due to a typo in https://github.com/cms-sw/cmssw/pull/37133/files#diff-f0c298feb52e2a455f0a07a4ea80c12b3b660e38e6737d177570ccc9a94775d3L147

perrotta commented:
Thank you @jfernan2 (indeed, now I remember that we discussed it already in the previous PR)
Since I was there, I updated the PR description with your suggestion myself

perrotta commented:
+1

NiharSaha commented:
> Thank you @jfernan2 (indeed, now I remember that we discussed it already in the previous PR) Since I was there, I updated the PR description with your suggestion myself

Thanks, @perrotta  for updating the PR description. I somehow missed @jfernan2's suggestion to update the description as I was traveling. Good to see that the PR is already merged. I guess I have implemented all the comments in this PR.
